airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/state/SourceStateIterator.java

```diff
@@ -19,16 +22,24 @@ public class SourceStateIterator<T> extends AbstractIterator<AirbyteMessage> imp
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SourceStateIterator.class);
   private final Iterator<T> messageIterator;
+  private final ConfiguredAirbyteStream stream;
+  private final StateEmitFrequency stateEmitFrequency;
   private boolean hasEmittedFinalState = false;
   private long recordCount = 0L;
   private Instant lastCheckpoint = Instant.now();
 
-  private final SourceStateIteratorManager sourceStateIteratorManager;
+  public record StateEmitFrequency(long syncCheckpointRecords, Duration syncCheckpointDuration) {}
```

Comment:
nit : can we just have a different file for this?

Reply:
done